### PR TITLE
HQ vehicle store UI rework

### PR DESCRIPTION
**Pull request by wersal454**
## What type of PR is this?
1. [ ] Bug
2. [x] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:

Attempt at adding more filtering into HQ vehicle store

### Please specify which Issue this PR Resolves (If Applicable).
"This PR closes #XXXX!"

### Please verify the following.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:
not worki at the moment (13.07.2024)

### CODE REVIEW

**Comment by wersal454** [comment]
Might need to make a port to community version to maybe receive help with this one

**Comment by wersal454** [comment]
done

File: A3A/addons/scrt/Encounter/fn_encounter_HeliSlingloadCargo.sqf
Comment (Silence-112) [inline A3A/addons/scrt/Encounter/fn_encounter_HeliSlingloadCargo.sqf:1]: was this change meant for another PR?

**Review by Silence-112 — CHANGES_REQUESTED** [review]

File: A3A/addons/scrt/Encounter/fn_encounter_HeliSlingloadCargo.sqf
Comment (wersal454) [reply A3A/addons/scrt/Encounter/fn_encounter_HeliSlingloadCargo.sqf:1]: no, it's just my hq is set near CSAT base, so I noticed that it wasn't working as intended, decided to fix it before I forget

**Review by Silence-112 — APPROVED** [review]

File: A3A/addons/gui/CfgFunctions.hpp [hunk 1]
```diff
@@ -8,6 +8,10 @@ class CfgFunctions {
             class airSupportTab {};
             class arsenalLimitsDialog {};
             class buyVehicleDialog {};
+            class buyVehicleDialogCivil {};
+            class buyVehicleDialogOther {};
+            class buyVehicleDialogRebel {};
+            class buyVehicleDialogStatics {};
```
Comment (Silence-112) [inline A3A/addons/gui/CfgFunctions.hpp]: None of these functions exist anymore

```suggestion
```

**Review by Silence-112 — CHANGES_REQUESTED** [review]
The UI is nice to use, very good

I think that the stringtables should be updated to capitalize the vehicle type, e.g: `Civilian Cars` rather than `Civilian cars`. It's minor but this way it will maintain visual consistency with the arms dealer

Also there's a minor issue with the CfgFunctions, it's a simple fix. See the review

**Review by Silence-112 — APPROVED** [review]
Looks fine now

**Pull request merged**